### DISCUSSION TITLE
URL with Anchor Reloads Storybook and Removes the Anchor – How Can I Prevent This?

### DISCUSSION THREAD

**Discussion by GuilhermeABRoncari** [storybookjs/storybook#29379] [Help]
### Summary

I'm using #React #NextJS

Hello! This is my first time posting here, so I hope I'm doing this right! 😄

I've built a Storybook application and created a custom full-text search system inside it, which works perfectly. However, when I create a URL with an anchor (linking to a specific subtitle in the documentation) and open that URL in a new tab or browser, Storybook always restarts. It doesn't automatically scroll to the subtitle, and it also removes the anchor from the URL.

I've noticed this behavior doesn't happen in other Storybook projects, like [Fluent UI](https://react.fluentui.dev/?path=/docs/concepts-introduction--docs), where URLs with anchors work as expected, and the page scrolls to the correct section.

I'm not sure if there's a configuration or setup that I'm missing, but I'd really appreciate any guidance on how to prevent Storybook from restarting and make it scroll to the correct section when opening URLs with anchors in a new tab.

Thanks in advance for your help!

### Additional information

Thats my main.ts:
```
import type { StorybookConfig } from '@storybook/nextjs';

const config: StorybookConfig = {
  stories: ['../src/**/*.mdx', '../src/**/*.stories.tsx'],
  addons: [
    '@storybook/addon-designs',
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-themes'
  ],
  framework: {
    name: "@storybook/nextjs",
    options: {}
  },
  staticDirs: ['../public'],
  webpackFinal: async (config) => {
    config.optimization = {
      ...config.optimization,
      splitChunks: {
        chunks: 'all',
      },
      minimize: true,
    };
    config.performance = {
      ...config.performance,
      maxEntrypointSize: 51200000,
      maxAssetSize: 51200000,
    }
    return config
  },
}

export default config
```

Version 8.3.5

**Comment by amyrlam** [comment]
I have the same issue. 

(1) Where in Storybook is autogenerating the anchor links for stories on a component page? 

(2) Not sure what code is stripping the anchor link when opening it in a new tab?
